kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt

```diff
@@ -110,17 +126,37 @@ class UploadSourceTask : SourceTask() {
                         }
                     }
 
-                    val result = converter.convert(record)
-                    result.result?.takeIf(List<*>::isNotEmpty)?.let {
-                        return@poll it
+                    try {
+                        val result = converter.convert(record)
+                        result.result?.takeIf(List<*>::isNotEmpty)?.let {
+                            return@poll it
+                        }
+                    } catch (exe: ConversionFailedException) {
+                        logger.error("Could not convert record ${record.id}", exe)
+                        updateRecordFailure(record)
                     }
+
                 }
             }
 
             Thread.sleep(pollInterval)
         }
     }
 
+    private fun updateRecordFailure(record: RecordDTO, reason: String? = "Could not convert this record. Please refer to the conversion logs for more details") {
+        logger.info("Update record conversion failure")
+        val metadata = uploadClient.retrieveRecordMetadata(record.id!!)
+        val updatedMetadata = uploadClient.updateStatus(record.id!!, metadata.copy(
+            status = "FAILED",
+            message = reason
+        ))
+
+        if(updatedMetadata.status == "FAILED") {
```

Comment:
This is already set the line above?

Reply:
Yes. It is mainly to make these request sequentially and verify if the updated record has the expected status